newrelic/resource_newrelic_workflow_test.go

```diff
@@ -0,0 +1,191 @@
+//go:build integration
+// +build integration
+
+package newrelic
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/newrelic/newrelic-client-go/pkg/ai"
+)
+
+func TestNewRelicWorkflow_Basic(t *testing.T) {
+	t.Skip("Skipping TestNewRelicWorkflow_Basic.  AWAITING FINAL IMPLEMENTATION!")
+
+	resourceName := "newrelic_workflow.test-workflow"
+	rand := acctest.RandString(5)
+	rName := fmt.Sprintf("tf-workflow-test-%s", rand)
+	enrichments := `{
+						nrql {
+						  name = "Log"
+						  configurations {
+						   query = "SELECT count(*) FROM Log"
+						  }
+						}
+					}`
+
+	destinationID := "4756c466-c29f-4f89-9cb4-382cabfcef61"
+	var channelId string
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccNewRelicWorkflowDestroy,
+		Steps: []resource.TestStep{
+			// Create channel
+			{
+				Config: testNewRelicNotificationChannelConfigByType(rName, "WEBHOOK", "IINT", destinationID, `{
+					key = "payload"
+					value = "{\n\t\"id\": \"test\"\n}"
+					label = "Payload Template"
+				}`),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNewRelicNotificationChannelExists(resourceName, channelId),
+				),
+			},
```

Comment:
not sure if this will work here @sanderblue  - how can I add here a channel creation?

Reply:
Sorry for the delay. If we want to create a notification channel to use in the test, we can add it to the test configurations in the test file by adding a notification resource and referencing it in the workflow config string. 

Something like this
```hcl
resource "newrelic_notification_destination" "foo" {
  name = "example-destination"
  type = "WEBHOOK"

  properties {
    // ..
  }

  auth = {
    // ..
  }
}

resource "newrelic_notification_channel" "foo" {
  destination_id = newrelic_notification_destination.foo.destination_id   # reference the destination 
  name = "webhook-example"
  type = "WEBHOOK"
  // .. etc

  properties {
    // ..
  }
}

resource "newrelic_workflow" "foo" {
  // ..

  enrichments {
    // ..
  }

  issues_filter {
    // ..
  }

  destination_configurations {
    channel_id = newrelic_notification_channel.foo.channel_id   # reference the channel 
  }
}
```